Add v4 to v5 migration guide for Cordova (#1062)
adjust/dev-docs: src/content/docs/sdk/cordova/migration/v4-to-v5.mdx

```jsx
adjustConfig.enableLinkMe();
```

### Only read device IDs once {#read-device-id-once}

In SDK v4, you can call the `setReadDeviceInfoOnceEnabled` method on your `adjustConfig` with the value `true` to instruct the SDK to only read device IDs once.

```jsx
adjustConfig.setReadDeviceInfoOnceEnabled(true);
```

In SDK v5, you need to call the `enableDeviceIdsReadingOnce` method on your `adjustConfig` to instruct the SDK to only read device IDs once. The default state is `false`.

```jsx
adjustConfig.enableDeviceIdsReadingOnce();
```

### Offline mode {#offline-mode}

In SDK v4, you can enable and disable offline mode the SDK by calling `Adjust.setOfflineMode` with a `bool` argument.

```jsx
Adjust.setOfflineMode(true);
Adjust.setOfflineMode(false);
```

In SDK v5, this feature is split into separate commands for clarity.

-  Call `Adjust.switchToOfflineMode` to set the SDK to offline mode.
-  Call `Adjust.switchBackToOnlineMode` to set the SDK back to online mode.

```jsx
Adjust.switchToOfflineMode(); // Put the SDK in offline mode
Adjust.switchBackToOnlineMode(); // Put the SDK back in online mode
```

### Session callback parameters {#callback-params}

> In SDK v4, you can add session parameters by passing a key-value pair to the `Adjust.addSessionCallbackParameter` method and remove individual parameters using the `Adjust.removeSessionCallbackParameter` method.

Reviewer: `session parameters` -> `session callback parameters`

```jsx
Adjust.addSessionCallbackParameter("key", "value");
Adjust.removeSessionCallbackParameter("key");
Adjust.resetSessionCallbackParameters();
```

> In SDK v5, session parameters are renamed to global parameters.

Reviewer: `session parameters` -> `session callback parameters`

   console.log(sessionFailure.message);
});
```

### Event success callbacks {#event-success}

In SDK v4, you can call the `setEventTrackingSucceededCallbackListener` to register a succeeded event recording by the SDK.

```jsx
adjustConfig.setEventTrackingSucceededCallbackListener(function (eventSuccess) {
   // Printing all event success properties.
   console.log("Event recording succeeded!");
   console.log(eventSuccess.message);
   console.log(eventSuccess.timestamp);
   console.log(eventSuccess.eventToken);
   console.log(eventSuccess.callbackId);
   console.log(eventSuccess.adid);
   console.log(eventSuccess.jsonResponse);
});
```

In SDK v5, this has been renamed to `setEventTrackingSucceededCallback`.

```jsx
adjustConfig.setEventTrackingSucceededCallback(function (eventSuccess) {
   // Printing all event success properties.
   console.log("Event recording succeeded!");
   console.log(eventSuccess.message);
   console.log(eventSuccess.timestamp);
   console.log(eventSuccess.eventToken);
   console.log(eventSuccess.callbackId);
   console.log(eventSuccess.adid);
   console.log(eventSuccess.jsonResponse);
});
```

### Event failure callbacks {#event-failure}

> In SDK v4, you can call the `setEventTrackingFailedCallbackListener` to register a succeeded event recording by the SDK.

Reviewer: `succeeded` -> `failed`

   price,
   currency,
   transactionId,
   receipt,
);
```

In SDK v5, you don't need to pass the `receipt` argument as it's no longer needed for purchase verification.

```jsx
var subscription = new AdjustAppStoreSubscription(
   price,
   currency,
   transactionId,
);
```

### Reattribution using deep links {#reattribution-using-deep-links}

In SDK v4, you can pass your deep link information to the `Adjust.appWillOpenUrl` method.

```jsx
Adjust.appWillOpenUrl("url");
```

In SDK v5, this has been renamed to `Adjust.processDeeplink` for clarity. A new `AdjustDeeplink` class has been added for constructing deep links. To process a deep link, instantiate an `AdjustDeeplink` object with your deep link URL and pass it to the `Adjust.processDeeplink` method.

```jsx
Adjust.processDeeplink(new AdjustDeeplink("url"));
```

### Launch deferred deep links {#launch-deferred-deeplinks}

In SDK v4, you can call the `setShouldlaunchDeeplink` method on your `adjustConfig` instance with the value `true` to enable the SDK to open deep links or `false` to disable the opening of deep links.

```jsx
adjustConfig.setShouldLaunchDeeplink(true);
```

> In SDK v5, deep links are opened by default. You need to call the `isDeferredDeeplinkOpeningEnabled` method on your `adjustConfig` instance to enable the SDK to open deep links.

Reviewer: `isDeferredDeeplinkOpeningEnabled` should be replaced with  `disableDeferredDeeplinkOpening`

```jsx
Adjust.processDeeplink("url", function (resolvedLink) {
   console.log("Resolved link: " + resolvedLink);
});
```

In SDK v5, you need to send an `AdjustDeeplink` object initialized with the deep link `url`. This returns the original unshortened deep link.

```jsx
Adjust.processAndResolveDeeplink(new AdjustDeeplink("url"), function(resolvedLink) {
   console.log("Resolved link: " + resolvedLink);
};
```

### Deep link callback listener {#deeplink-callback-listener}

In SDK v4, you can configure a listener function that fires when a deep link is received by calling the `setDeferredDeeplinkCallbackListener` method on your `AdjustConfig` instance.

```jsx
adjustConfig.setDeferredDeeplinkCallbackListener(function (deeplink) {});
```

In SDK v5, this method has been renamed to `setDeferredDeeplinkCallback`. All arguments remain the same.

```jsx
adjustConfig.setDeferredDeeplinkCallback(function (deeplink) {});
```

### COPPA compliance {#coppa-compliance}

In SDK v4, you can call the `coppaCompliantEnabled` method on your `adjustConfig` instance with the value `true` to enable COPPA compliance.

```jsx
adjustConfig.setCoppaCompliantEnabled(true);
```

> In SDK v5, you need to call the `isCoppaComplianceEnabled` method on your `adjustConfig` instance to enable COPPA compliance. The default state is `false`.

Reviewer: `isCoppaComplianceEnabled` -> `enableCoppaCompliance`

```jsx
adjustConfig.setPlayStoreKidsAppEnabled(true);
```

In SDK v5, you need to call the `enablePlayStoreKidsCompliance` method of your `adjustConfig` instance to enable compliance. The default state is `false`.

```jsx
adjustConfig.enablePlayStoreKidsCompliance();
```

### Set data residency and URL strategy {#url-strategy}

In SDK v4, URL strategy and data residency domains are defined as constants in the `adjustConfig` class.

```jsx
adjustConfig.setUrlStrategy(AdjustConfig.DataResidencyEU);
```

In SDK v5, you need to pass your chosen domain or domains as an array. You need to also set the following:

-  `useSubdomains` (`bool`): Whether the domain should be treated as an Adjust domain. If `true`, the SDK will prefix the domains with Adjust-specific subdomains. If `false`, the SDK will use the provided domain as-is, without adding any prefixes.
-  `isDataResidency` (`bool`): Whether the domain should be used for data residency.

```jsx
adjustConfig.setUrlStrategy(["eu.adjust.com"], true, true);
```

Check the table below to see how to configure your URL strategy in SDK v5.

> <Table autoWidth="1,2">

Reviewer: Could we change the table layout to accommodate the domain strings better?
![image](https://github.com/user-attachments/assets/5e593837-a8ae-47c8-b2c4-d02d9efc1a88)

Author: @genadyb Looks like something changed with Atlas. I have a plan to fix this, but it's being handled in a different PR.

```

This setting has been removed in SDK v5 since now only the final attribution will be delivered through the attribution callback on the Android platform.

### Set referrer {#set-referrer}

In SDK v4, you can call the `setReferrer` method to trigger an SDK click with a custom click ID when the SDK starts.

```jsx
Adjust.setReferrer("referrer");
```

This method has been removed in SDK v5. Google has deprecated the `INSTALL_REFERRER` intent method of delivering referrer information for Google Play Services. If you currently use this method, migrate to the Google Play Referrer API.

### Convert universal link {#convert-universal-link}

In SDK v4, you can call the `convertUniversalLink` method.

```jsx
Adjust.convertUniversalLink(function (url, scheme, callback) {});
```

This method has been removed in SDK v5.

### AdjustEvent changes {#adjustevent-changes}

In SDK v4, you can send a `receipt` to the `adjustEvent.setReceipt` method on your `adjustEvent` for subscriptions and purchases.

```jsx
adjustEvent.setReceipt("receipt");
```

This parameter has been removed in SDK v5 since it's no longer needed for purchase verification.

### Purchase verification {#purchase-verification}

> In SDK v4, you must instantiate an `AdjustAppStorePurchase` object with a `receipt` property. This property is used for purchase verification.

Reviewer: What does fit better here `Property` or` argument` term?

```js
Adjust.checkForNewAttStatus();
```

This method has been removed in SDK v5.

### Final Android attribution {#final-android-attribution}

In SDK v4, you can call the `setFinalAndroidAttributionEnabled` method on your `adjustConfig` instance with the value `true` to deliver final Android attribution.

```js
adjustConfig.setFinalAndroidAttributionEnabled(true);
```

This setting has been removed in SDK v5 since now only the final attribution will be delivered through the attribution callback on the Android platform.

### Set referrer {#set-referrer}

In SDK v4, you can call the `setReferrer` method to trigger an SDK click with a custom click ID when the SDK starts.

```js
Adjust.setReferrer("referrer");
```

This method has been removed in SDK v5. Google has deprecated the `INSTALL_REFERRER` intent method of delivering referrer information for Google Play Services. If you currently use this method, migrate to the Google Play Referrer API.

### Convert universal link {#convert-universal-link}

In SDK v4, you can call the `convertUniversalLink` method.

```js

> Adjust.convertUniversalLink(function (url, scheme, callback) {});

Reviewer 2: ```suggestion
Adjust.convertUniversalLink(url, scheme, function(convertedLink) {
    console.log("Converted link = " + convertedLink);
});
```

```

This method has been removed in SDK v5.

### AdjustEvent changes {#adjustevent-changes}

> In SDK v4, you can send a `receipt` to the `adjustEvent.setReceipt` method on your `adjustEvent` for subscriptions and purchases.

Reviewer 2: ```suggestion
In SDK v4, you can send a `receipt` to the `adjustEvent.setReceipt` method on your `AdjustEvent` instance for purchase verification.
```

```js
adjustEvent.setReceipt("receipt");
```

This parameter has been removed in SDK v5 since it's no longer needed for purchase verification.

### Purchase verification {#purchase-verification}

In SDK v4, you must instantiate an `AdjustAppStorePurchase` object with a `receipt` argument. This value is used for purchase verification.

```js
var appStorePurchase = new AdjustAppStorePurchase(
   "receipt",
   "product-id",

>    "transaction-id",

Reviewer 2: ```suggestion
   "transaction-id"
```
Not wrong, just styling.

);
```

This property has been removed in SDK v5. If you're using the `AdjustAppStorePurchase` class for purchase verification, you must remove this argument from your construction call.

```js
var appStorePurchase = new AdjustAppStorePurchase(
   "product-id",

>    "transaction-id",

Reviewer 2: ```suggestion
   "transaction-id"
```
Not wrong, just styling.
